pkg/cli/admin/release/bug.go

```diff
@@ -0,0 +1,309 @@
+package release
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+type BugType int
+
+const (
+	Bugzilla BugType = iota
+	Jira
+)
+
+const (
+	maxRetries = 2
+
+	bugzillaBrowseURL = "https://bugzilla.redhat.com/show_bug.cgi?id=%d"
+	bugzillaRestURL   = "https://bugzilla.redhat.com/rest/bug"
+	jiraBrowseURL     = "https://issues.redhat.com/browse/OCPBUGS-%d"
+	JiraRestURL       = "https://issues.redhat.com/rest/api/latest/issue/OCPBUGS-%d"
+)
+
+var (
+	reBugzillaBug = regexp.MustCompile(`^[B|b]ug(?:s)?\s([\d\s,]+)(?:-|:)\s*`)
+	reJiraBug     = regexp.MustCompile(`^OCPBUGS-([\d\s,]+)(?:-|:)\s*`)
+)
+
+// BugList stores Bug list
+type BugList struct {
+	Bugs []Bug
+}
+
+// Bug stores referenced bug ids and
+// source of bug type like Bugzilla or Jira.
+type Bug struct {
+	ID     int
+	Source BugType
+}
+
+// BugRemoteList stores BugRemoteInfo list
+type BugRemoteList struct {
+	Bugs []BugRemoteInfo `json:"bugs"`
+}
+
+// BugRemoteInfo stores the detail information of bug retrieved
+// from remote url like bugzilla or jira.
+type BugRemoteInfo struct {
+	ID       int    `json:"id"`
+	Status   string `json:"status"`
+	Priority string `json:"priority"`
+	Summary  string `json:"summary"`
+	Source   BugType
```

Comment:
From the reviewer's point of view, to make review easy, this PR might be split into different commits. However, this `Source` field, which is used to discriminate bugs according to it's source, demands one commit by changing everywhere at one time.